src/operator/nn/softmax-inl.h

```diff
@@ -36,22 +36,22 @@ namespace op {
 namespace mxnet_op {
 
 struct softmax_fwd {
-  template<typename DType>
-  MSHADOW_XINLINE static DType Map(DType a, DType b) {
+  template<typename DType, typename AType>
+  MSHADOW_XINLINE static DType Map(DType a, AType b) {
     return DType(expf(a)/b);
   }
 };
 
 
 struct log_softmax_fwd {
-  template<typename DType>
-  MSHADOW_XINLINE static DType Map(DType a, DType b) {
+  template<typename DType, typename AType>
+  MSHADOW_XINLINE static DType Map(DType a, AType b) {
     return DType(a - logf(b));
```

Comment:
Same here.

Reply:
Thanks. Good point. The same should apply to the gradient kernels